nixos/modules/services/networking/heyefi.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.heyefi;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.heyefi = {
+
+      cardMacaddress = mkOption {
+        default = "";
+        description = ''
+          An Eye-Fi card MAC address.
+          '';
+      };
+
+      uploadKey = mkOption {
+        default = "";
+        description = ''
+          An Eye-Fi card's upload key.
+          '';
+      };
+
+      uploadDir = mkOption {
+        example = /home/ryantm/picture;
+        description = ''
+          The directory to upload the files to.
+          '';
+      };
+
+      enable = mkOption {
+        default = false;
+        description = ''
+          Enable heyefi, an Eye-Fi upload server.
+        '';
+      };
+
+      user = mkOption {
+        default = "root";
```

Comment:
Don't forget to add ids to https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/ids.nix